WhateverGreen/kern_rad.cpp

```diff
@@ -1081,3 +1090,22 @@ bool RAD::wrapNotifyLinkChange(void *atiDeviceControl, kAGDCRegisterLinkControlE
 
 	return ret;
 }
+
+void RAD::updateGetHWInfo(IOService *accelVideoCtx, void *hwInfo) {
+	IOService *accel, *pciDev;
+	accel = OSDynamicCast(IOService, accelVideoCtx->getParentEntry(gIOServicePlane));
+	if (accel == NULL) {
+		SYSLOG("rad", "getHWInfo: no parent found for accelVideoCtx!");
+		return;
+	}
+	pciDev = OSDynamicCast(IOService, accel->getParentEntry(gIOServicePlane));
+	if (pciDev == NULL) {
+		SYSLOG("rad", "getHWInfo: no parent found for accel!");
+		return;
+	}
+	uint16_t org = *(uint16_t *)((char *)hwInfo + 0x4);
```

Comment:
Please use `getMember` from `kern_util.hpp` here and below. `getMember<uint16_t>(hwInfo, 0x4)` will be a lot more readable.

Reply:
You should also try and compute struct member offsets in runtime rather than hardcode them for better compatibility (see https://github.com/acidanthera/NVMeFix/blob/master/NVMeFix/NVMeFix.cpp#L76, https://github.com/acidanthera/NVMeFix/blob/master/NVMeFix/NVMeFixPlugin.hpp#L159 for example.) Of course, this only makes sense to do when you believe the code accessing the relevant struct member is less volatile than the struct definition itself, and the mov is not that hard to find, etc.

Reply:
The struct should be stable as it's the second member of a pretty large struct. The contents themselves defines all sorts of different values for various GPUs so I think it's harder.